### ISSUE TITLE
Network setting doesn't follow proxy settings for web access in Windows 10

### ISSUE THREAD

**Issue by qgib** [qgis/QGIS#28034]
Author Name: **Christoph Candido** (Christoph Candido)
Original Redmine Issue: [20213](https://issues.qgis.org/issues/20213)
Affected QGIS version: 3.6.0
Redmine category:network

---
QGIS 3.3.0-Master doesn't follow it's own proxy settings for web access in Windows 10 Enterprise.
It only follows the general proxy settings in Windows.

We cannot disable the proxy in QGIS, if the proxy is allready configured in Windows (Local Area Network Settings -> Proxy Server).
(The Windows environment variables http_proxy,https_proxy,no_proxy don't help and are not defined)

As a result of the problem, we cannot open local WMS services inside our network.
The WMS request allways runs over our central proxy with NTLM authentication, which QGIS refuses to handle.

---

- [Capture.JPG](https://issues.qgis.org/attachments/download/14602/Capture.JPG) (Christoph Candido)

**Comment by FERRATON** [comment]
According to my tests, it is possible that if a URL is in the exclusion list, QGIS will try to use the system proxy instead of not using a proxy (direct method).

Maybe see how the fallbackproxy variable is powered in 
https://github.com/qgis/QGIS/blob/master/src/core/qgsnetworkaccessmanager.cpp (line 73,...)
(Sorry if this is not a good track, I'm not a developer....)

**Comment by troopa81** [comment]
Proposed PR https://github.com/qgis/QGIS/pull/30450

**Comment by almaniac** [comment]
I've been having the same issue with 3.8.3.  The only way I've been able to get QGIS to talk to WMS servers on my own network is to disable the proxy through Internet Options under IE.  This, of course, then disables my connection to the Internet, which must go through the proxy.

I've tried configuring the proxy in QGIS, and ensuring there were exceptions for all the WMS servers I'm trying to use (with Ineternet Options and with QGIS internally) with no result.
